src/Symfony/Component/Console/Command/Command.php

```diff
@@ -273,12 +267,25 @@ public function run(InputInterface $input, OutputInterface $output): int
 
         $input->validate();
 
+        $lockName = $this->resolveLockName($input);
+
+        if (null !== $lockName) {
+            $lockHandler = new LockHandler();
+            if (!$lockHandler->lock($lockName, self::isBlockingLock())) {
+                return self::FAILURE;
+            }
+        }
+
         if ($this->code) {
             $statusCode = ($this->code)($input, $output);
         } else {
             $statusCode = $this->execute($input, $output);
         }
 
+        if (null !== $lockName) {
+            $lockHandler->release($lockName);
```

Comment:
We should have a way to release early the command instead of waiting the end of the command.

Reply:
Now, in every command you can call `$this->releaseLock($input)` to release the lock at any moment you want.